src/azure-cli/azure/cli/command_modules/role/_params.py

```diff
@@ -120,7 +120,7 @@ def load_arguments(self, _):
         c.argument('identifier_uri', help='graph application identifier, must be in uri format')
         c.argument('spn', help='service principal name')
         c.argument('upn', help='user principal name, e.g. john.doe@contoso.com')
-        c.argument('query_filter', options_list=['--filter'], help='OData filter')
+        c.argument('query_filter', options_list=['--filter'], help='OData filter, e.g. --filter \"displayname eq \'test\' and servicePrincipalType eq \'Application\'\"')
```

Comment:
We don't need to escape double quote within single quoted string. 

```suggestion
        c.argument('query_filter', options_list=['--filter'], help='OData filter, e.g. --filter "displayname eq \'test\' and servicePrincipalType eq \'Application\'"')
```